src/cunumeric/matrix/potrf.cc

```diff
@@ -25,44 +25,47 @@ namespace cunumeric {
 using namespace Legion;
 using namespace legate;
 
-template <typename Potrf, typename VAL>
-static inline void potrf_template(Potrf potrf, VAL* array, int32_t m, int32_t n)
-{
-  char uplo    = 'L';
-  int32_t info = 0;
-  potrf(&uplo, &n, array, &m, &info);
-  if (info != 0) throw legate::TaskException("Matrix is not positive definite");
-}
-
 template <>
 struct PotrfImplBody<VariantKind::CPU, LegateTypeCode::FLOAT_LT> {
   void operator()(float* array, int32_t m, int32_t n)
   {
-    potrf_template(LAPACK_spotrf, array, m, n);
+    char uplo    = 'L';
+    int32_t info = 0;
+    LAPACK_spotrf(&uplo, &n, array, &m, &info);
```

Comment:
Would it not be possible to just remove the first argument from `potrf_template` and have it always invoke `LAPACK_spotrf` directly?